versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/indexes/StoreIndexImpl.java

```diff
@@ -479,7 +479,23 @@ private ByteBuffer serializeKey(
 
   static <V> StoreIndex<V> deserializeStoreIndex(ByteBuffer serialized, ElementSerializer<V> ser) {
     byte version = serialized.get();
-    checkArgument(version == 1, "Unsupported serialized representation of KeyIndexSegment");
+    boolean needSort = false;
+    switch (version) {
+      case 1:
+        // In PR #6743, introducing min/max/prefix-key arguments for "get entries" and "get diff",
+        // StoreKey.compareTo() was updated, which might lead to a different order in the elements
+        // list. The "current" version has been bumped, version=1 serialized indexes are sorted
+        // after they have been loaded.
```

Comment:
Do you have an example where the sort order actually changes? I could not figure one out mentally :)

Reply:
ouch - again